### PR TITLE
Fix sort class bug

### LINKED ISSUE #239: [PE-D][Tester A] Sort based on class time not accurate
Nov before Oct

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: LuYiting0913/ped#13

### LINKED ISSUE #260: [PE-D][Tester D] Sorting the student list in ascending order does not sort by the class date properly
According to the user guide, it seems like `sort class` should sort the students in order of who's class is the closest to the current time, but it places the class that is next week first.

[teachersPet.txt](https://raw.githubusercontent.com/ngxingyu/ped/main/files/8ced3cb8-d010-4f5c-bece-50db0a847d07.txt)

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: ngxingyu/ped#6

### PR DESCRIPTION
**Pull request by Echomo-Xinyu**
Closes #239
Closes #260

### CODE REVIEW

**Review by seanflyyy — APPROVED** [review]
LGTM! Seems like there was a case of the dates being on different days that was missed out on. After that was rectified, it's working as intended. Good job!

File: src/main/java/seedu/address/model/student/Class.java [hunk 1]
```diff
@@ -332,11 +332,23 @@ public boolean equals(Object other) {
...
      */
     public int compareToByStartTime(Class aclass) {
-        requireAllNonNull(this.startTime, aclass.startTime);
+        requireAllNonNull(this.date, this.startTime, aclass.date, aclass.startTime);
+        assert this.date.equals(aclass.date);
         return this.startTime.compareTo(aclass.startTime);
     }
 
+    /**
+     * Returns 1 is the {@code Class} starts before the given {@code aclass}.
```
Comment (gnahzeus) [inline src/main/java/seedu/address/model/student/Class.java:344]: Do you mean `Returns 1 if ...`?

**Review by gnahzeus — COMMENTED** [review]
Minor typo in one of the comments

File: src/main/java/seedu/address/model/student/Class.java (hunk 1 above)
Comment (Echomo-Xinyu) [reply src/main/java/seedu/address/model/student/Class.java:344]: Yes and here it should be `returns -1 if ...`. Thanks for bringing it up!

**Pull request merged**